src/makeInfinityAdder.js

```diff
@@ -3,8 +3,24 @@
 /***
  * @return {function}
  */
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(currentNum) {
+  let amount = currentNum || 0;
+
+  function sum(nextNum) {
+    if (nextNum !== undefined) {
+      amount += nextNum;
+
+      return sum;
+    } else {
+      const result = amount;
+
+      amount = 0;
```

Comment:
Resetting `amount` to 0 here might cause issues if the user wants to continue adding numbers after retrieving the sum. Consider whether this behavior is intended or if `amount` should retain its value for subsequent additions.